service/service.go

```diff
@@ -196,19 +198,63 @@ func New(config Config) (*Service, error) {
 		}
 	}
 
-	var metricsCollector *collector.Collector
+	var newHelper *collector.Helper
 	{
-		c := collector.Config{
+		c := collector.HelperConfig{
 			G8sClient: g8sClient,
 			K8sClient: k8sClient,
 			Logger:    config.Logger,
 
-			AwsConfig:             awsConfig,
+			AwsConfig: awsConfig,
+		}
+
+		newHelper, err = collector.NewHelper(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var legacyCollector *collector.Collector
+	{
+		c := collector.Config{
+			Helper: newHelper,
+			Logger: config.Logger,
+
 			InstallationName:      config.Viper.GetString(config.Flag.Service.Installation.Name),
 			TrustedAdvisorEnabled: config.Viper.GetBool(config.Flag.Service.AWS.TrustedAdvisor.Enabled),
 		}
 
-		metricsCollector, err = collector.New(c)
+		legacyCollector, err = collector.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var elbCollector *collector.ELBCollector
+	{
+		c := collector.ELBConfig{
+			Helper: newHelper,
+			Logger: config.Logger,
+
+			InstallationName: config.Viper.GetString(config.Flag.Service.Installation.Name),
+		}
+
+		elbCollector, err = collector.NewELB(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
```

Comment:
Maybe it's an idea to have a `NewCollectorSet` inside `collector package? We wouldn't have to expose this `Helper` type. I also think less initializations in this file is beneficial.

Reply:
I did this.